reviewing/list/src/pruneTree.ts

```diff
@@ -0,0 +1,78 @@
+import { type DependenciesHierarchy, type PackageNode } from '@pnpm/reviewing.dependencies-hierarchy'
+import { type PackageDependencyHierarchy } from './types'
+import { createHash } from 'crypto'
+
+export function pruneDependenciesTrees (trees: PackageDependencyHierarchy[] | null, limit: number): PackageDependencyHierarchy[] {
+  if (trees === null) {
+    return []
+  }
+
+  return trees.map((tree) => {
+    const endLeafPaths: PackageNode[][] = []
+    const visitedNodes = new Set<string>()
+
+    function findEndLeafs (node: PackageNode, path: PackageNode[]): void {
```

Comment:
The plural of leaf is leaves
```suggestion
    function findEndLeaves (node: PackageNode, path: PackageNode[]): void {
```